core/domain/acl_decorators.py

```diff
@@ -123,9 +39,15 @@ def can_play_exploration(handler):
     """Decorator to check whether user can play given exploration."""
 
     def test_can_play(self, exploration_id, **kwargs):
-        if check_activity_accessible(
+        if exploration_id in feconf.DISABLED_EXPLORATION_IDS:
+            raise self.PageNotFoundException
+
+        exploration_rights = rights_manager.get_exploration_rights(
+            exploration_id, strict=False)
+
+        if rights_manager.check_can_access_activity(
                 self.user_id, self.actions, feconf.ACTIVITY_TYPE_EXPLORATION,
-                exploration_id):
+                exploration_rights):
```

Comment:
Won't adding extra parameter to request through handlers make thing a bit indirect to understand? I mean now handlers will accept two arguments while in main.py we only show one argument. This doesn't look clean to me.